aiida/backends/tests/test_restapi.py

```diff
@@ -69,14 +73,37 @@ def setUpClass(cls, *args, **kwargs):
 
         resources = {'num_machines': 1, 'num_mpiprocs_per_machine': 1}
 
+        calcfunc = orm.CalcFunctionNode(computer=cls.computer)
+        calcfunc.store()
+
         calc = orm.CalcJobNode(computer=cls.computer)
         calc.set_option('resources', resources)
         calc.set_attribute("attr1", "OK")
         calc.set_attribute("attr2", "OK")
 
         calc.add_incoming(structure, link_type=LinkType.INPUT_CALC, link_label='link_structure')
         calc.add_incoming(parameter1, link_type=LinkType.INPUT_CALC, link_label='link_parameter')
+
+        aiida_in = 'The input file\nof the CalcJob node'
+        # Add the file to the CalcJobNode folder calcjob_inputs
+        tmpdir_in = tempfile.mkdtemp()
```

Comment:
If you are only writing a single file, it is perhaps nicer to use the `put_object_from_filelike` that was designed for such use cases. You can then do something like:
```
with tempfile.TemporaryFile() as handle:
     handle.write()
     handle.flush()
     handle.seek(0)
     calc.put_object_from_filelike(handle, key='calcjob_inputs', force=True)
```
This way you do not need to create a whole directory.

Reply:
done, thanks!